include/ignition/gazebo/components/Visual.hh

```diff
@@ -81,11 +81,21 @@ namespace components
   IGN_GAZEBO_REGISTER_COMPONENT("ign_gazebo_components.Visual", Visual)
 
   /// \brief A component that contains a visual plugin's SDF element.
+  /// \note This will be deprecated in Gazebo 7 (Garden), please the use
+  /// sdf::Plugins interface.
   using VisualPlugin = Component<sdf::ElementPtr,
                                  class VisualPluginTag,
                                  serializers::SdfElementSerializer>;
   IGN_GAZEBO_REGISTER_COMPONENT("ign_gazebo_components.VisualPlugin",
       VisualPlugin)
+
+  /// \brief A component that contains a visual plugin's SDF Plugin objects.
+  using VisualPlugins = Component<sdf::Plugins,
```

Comment:
This looks like a duplication of the `SystemInfo` component being introduced in #1404. I was planning to deprecate `VisualPlugins` after that PR was merged.

The main difference between `SystemInfo` and `VisualPlugins` is that one uses a `msgs::Plugin_V` and the other `sdf::Plugins`. In general, using messages in components is good for serialization. The downside is having to convert the message to SDF in the receiving side when needed.

Reply:
Using `msgs` or `sdf` is fine with me. I've been making changes to components with the assumption that I should be using SDF in order to keep protobuf out of public headers. Sounds like you are doing the opposite.

Now seems like a good time to resolve #494 and pick a storage type.

Reply:
> Now seems like a good time to resolve https://github.com/ignitionrobotics/ign-gazebo/issues/494 and pick a storage type.

That sounds like more work than we can do on this PR and #1404. I'd prefer going with messages for now because they're lighter and more easily serialized. Especially for plugins, which only hold strings (no math), messages should be more convenient than SDF. This is setting aside all the issues with exposing messages in our public APIs.

Reply:
I've added some more [thoughts on this topic](https://github.com/ignitionrobotics/ign-gazebo/issues/494#issuecomment-1118902403).

Reply:
How about leaving Plugin information as sdf instead of msgs? Plugin information is transmitted infrequently (I think?) so the serialization overhead should be minor. If/when sdf::Plugin's API changes then we won't have to make changes to msgs or the converter. I think msgs::Plugin is currently not used in Gazebo.

Reply:
Since there are still issues serializing `sdf::Plugin` which prevent it from being used in a component, I'd like to move forward with using messages as proposed in #191.  

I'd like not to block these PRs on resolving #494, because that issue goes beyond components - it's about reducing the number of data storage classes that we have across all our libraries. For components, currently the best solution for Fortress are messages.

Reply:
The `VisualPlugins` component has been removed in: [c3dbeb7](https://github.com/gazebosim/gz-sim/pull/1352/commits/c3dbeb77ce46723301faf05927c9069dc1c6a1be)

I've also updated the sdf::Plugin stream extraction in your PR: https://github.com/gazebosim/sdformat/pull/1021